dali/pipeline/operator/operator.h

```diff
@@ -99,9 +99,12 @@ class DLL_PUBLIC OperatorBase {
   /**
    * @brief Setup of the operator - to be implemented by derived op.
    *
+   * In the setup stage, the operator can determine the shapes and types of the outputs.
+   * If it does, it can request that the executor allocates the output buffers for it.
+   *
    * @param output_desc describe the shape and type of the outputs (for the whole batch)
    * @param ws
-   * @return true iff the operator specified the output shape and type
+   * @return Whether the caller should provide buffers for the outputs.
    */
   virtual bool SetupImpl(std::vector<OutputDesc> &output_desc, const Workspace &ws) = 0;
```

Comment:
If we are doing this, we probably should split the request of allocation and the shape inference as well. For operators that don't want to request the output allocation most of them can still infer the output shapes based on the input shapes.
From time to time we talk about shape and type inference, and this is yet another thing independent from allocation.